src/main/java/io/vertx/core/impl/VertxImpl.java

```diff
@@ -871,11 +870,17 @@ private class InternalTimerHandler implements Handler<Void>, Closeable, Runnable
     private final ContextInternal context;
     private final java.util.concurrent.Future<?> future;
 
-    InternalTimerHandler(long timerID, Handler<Long> runnable, boolean periodic, long delay, ContextInternal context) {
+    InternalTimerHandler(long timerID,
+      Handler<Long> runnable,
+      boolean periodic,
+      long delay,
+      ContextInternal context,
+      ConcurrentMap<Long, InternalTimerHandler> timeouts) {
       this.context = context;
       this.timerID = timerID;
       this.handler = runnable;
       this.periodic = periodic;
+      timeouts.put(timerID, this);
```

Comment:
I think this should be left out of the constructor and we should instead move the timer scheduling outside of the constructor, to avoid passing `this` in a constructor.

Reply:
I have posted a version that moves the timer scheduling outside the constructor.